chapters/synchronous.tex

```diff
@@ -159,22 +156,25 @@ \subsection{Rationale for Clocked Semantics}\label{rationale-for-clocked-semanti
   equations, clocked controllers with continuous-time models can be
   directly defined in Modelica.
 \item
-  At a sample instant, an event iteration occurs (as for any other
-  event). A clocked partition, as well as a when-clause with a
-  \lstinline!sample! is evaluated exactly once at such an event instant.
-  However, the continuous-time model to which the sampled data
-  controller is connected, will be evaluated several times when the
-  overall system is simulated. With when-clauses, the continuous-time
-  part is typically evaluated three times at a sample instant (once,
-  when the sample instant is reached, once to evaluate the continuous
-  equations at the sample instant, and once when an event iteration
-  occurs since a discrete variable \lstinline!v! is changed and \lstinline!pre(v)!
-  appears in the equations). With clocked equations, no event iteration
-  is triggered if a clocked variable \lstinline!v! is changed and
-  \lstinline!previous(v)! appears in the equations, because the event
-  iteration cannot change the value of \lstinline!v!. As a result, typically the
-  simulation model is evaluated twice at a sample instant and therefore
-  the simulation is more efficient with clocked equations.
+  Clocked equations are straightforward to optimize because they are
+  evaluated exactly once at each an event instant.
+  In contrast a standard when-clause with \lstinline!sample! conceptually
+  require several evaluations of the model (in some cases tools
+  can optimize this to avoid unneeded evaluations).
+  The problem for the standard when-clause is that after \lstinline!v!
+  is changed, \lstinline!pre(v)! shall be updated and the model re-evaluated,
+  since the equations could depend on \lstinline!pre(v)!.
+  For clocked equations this iteration can be omitted
+  since \lstinline!previous(v)! can only occur in the clocked equations
+  that are only run the first event iterations.
```

Comment:
This sounds a bit contradictory; should _event iterations_ really be in the plural?

Reply:
ok